Garbage Collect old rendered MachineConfigs (#74675)
openshift/openshift-docs: modules/machineconfig-garbage-collect-removing.adoc

// Module included in the following assemblies:
//
// * post_installation_configuration/machine-configuration-tasks.adoc

:_mod-docs-content-type: PROCEDURE
[id="machineconfig-garbage-collect-removing_{context}"]
= Removing unused rendered machine configs

You can remove unused rendered machine configs by running the `oc adm prune renderedmachineconfigs` command with the `--confirm` command. For example, the command in the following procedure would delete the two oldest unused rendered machine configs in the `worker` machine config pool. If any rendered machine config is not deleted, the command output indicates which was not deleted and lists the reason for skipping the deletion.

.Procedure 

* List the rendered machine configs that can be automatically removed. The following command runs in dry-run mode, and no machine configs are removed. 
+
[source,terminal]
----
$ oc adm prune renderedmachineconfigs --pool-name=worker
----

> +

Reviewer: ```suggestion
+
The  command runs in dry-run mode, and no machine configs are removed. 
```

Author: Moved closer to the command.

.Example output
[source,terminal]
----
Dry run enabled - no modifications will be made. Add --confirm to remove rendered machine configs.
DRY RUN: Deleted rendered MachineConfig rendered-worker-23d7322831a57f02998e7e1600a0865f
DRY RUN: Deleted rendered MachineConfig rendered-worker-fc94397dc7c43808c7014683c208956e
DRY RUN: Skipping deletion of rendered MachineConfig rendered-worker-ad5a3cad36303c363cf458ab0524e7c0 as it's currently in use
----
+
--
where:

pool-name:: Optional: Specifies the machine config pool where you want to delete the machine configs from.
--

> * Remove unused rendered machine configs. Use the `--confirm` flag to remove the eligible machine configs.

Reviewer: ```suggestion
* Remove the unused rendered machine configs by running the following command:
```

+
[source,terminal]
----
$ oc adm prune renderedmachineconfigs --pool-name=worker --count=2 --confirm
----
+
--
where: 

> count:: Optional: Specifies the maximum number of unused rendered machine configs to delete, oldest first.

Reviewer: ```suggestion
`--count`:: Optional: Specifies the maximum number of unused rendered machine configs you want to delete, starting with the oldest.
```

> confirm:: Optional: Specifies that the MCO should delete the specific unused rendered machine configs. If not set, the command operates in dry-run mode and no configs are deleted.

Reviewer: ```suggestion
`--confirm`:: Optional: Specifies that the MCO should delete the specific unused rendered machine configs. Use this flag to remove the eligible machine configs. If not specified, the command operates in a dry-run mode and no configs are deleted.
```

> pool-name:: Optional: Specifies the machine config pool where you want to delete the machine configs from. If not set, all pools will be evaluated.

Reviewer: ```suggestion
`--pool-name`:: Optional: Specifies the machine config pool from which you want to delete the machine. If not specified, all the pools are evaluated.
```
